llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -519,6 +537,87 @@ SDValue XtensaTargetLowering::LowerImmediate(SDValue Op,
   return Op;
 }
 
+SDValue XtensaTargetLowering::LowerGlobalAddress(SDValue Op,
+                                                 SelectionDAG &DAG) const {
+  const GlobalAddressSDNode *G = cast<GlobalAddressSDNode>(Op);
+  SDLoc DL(Op);
+  auto PtrVt = getPointerTy(DAG.getDataLayout());
+  const GlobalValue *GV = G->getGlobal();
+
+  // Check Op SDNode users
```

Comment:
This handling doesn't make sense to me and you shouldn't depend on the user context. If you want to fold the global address into an immediate call operand, you can do that in your call handling

Reply:
Thank you very much for comment! I removed redundant code.